### PR TITLE
fix: More Backup Restore Fixes

### LINKED ISSUE #2857: [BUG] - "Something Went Wrong!" trying to login after database restore
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I was running 1.0.0beta as a TrueCharts app. They changed a bunch of stuff and I was no longer able to upgrade from that version to the 1.0.0 release as they published it on a different channel/train. But when I restore the database on the new deployment, I can never login again, with "Something Went Wrong!" message banner at the top. I am using the user/pass that was saved in my password manager. The default user/pass also no longer works after the backup. I have used the python reset password script and it also errors out.

### Steps to Reproduce

1. Deploy 1.0.0
2. Login as default user
3. Upload backup from 1.0.0b
4. Restore uploaded backup
5. Attempt to login again

### Please provide relevant logs

Trying to reset password with the python scripts causes a database error:
```
$ python change_password.py
Please enter the email of the user you want to reset: jacobmcdonald75023@gmail.com
Traceback (most recent call last):
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1965, in _exec_single_context
    self.dialect.do_execute(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 921, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column users.auth_method does not exist
LINE 1: ...ame, users.username, users.email, users.password, users.auth...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/mealie/scripts/change_password.py", line 40, in <module>
    main()
  File "/app/mealie/scripts/change_password.py", line 17, in main
    user = repos.users.get_one(confirmed, "email")
  File "/app/mealie/repos/repository_generic.py", line 159, in get_one
    result = self.session.execute(q).unique().scalars().one_or_none()
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2246, in execute
    return self._execute_internal(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2141, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
    result = conn.execute(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1412, in execute
    return meth(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 483, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1635, in _execute_clauseelement
    ret = self._execute_context(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1844, in _execute_context
    return self._exec_single_context(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1984, in _exec_single_context
    self._handle_dbapi_exception(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2339, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1965, in _exec_single_context
    self.dialect.do_execute(
  File "/opt/pysetup/.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 921, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column users.auth_method does not exist
LINE 1: ...ame, users.username, users.email, users.password, users.auth...
                                                             ^

[SQL: SELECT users.id, users.full_name, users.username, users.email, users.password, users.auth_method, users.admin, users.advanced, users.group_id, users.cache_key, users.login_attemps, users.locked_at, users.can_manage, users.can_invite, users.can_organize, users.owned_recipes_id, users.created_at, users.update_at, groups_1.id AS id_1, groups_1.name, groups_1.slug, groups_1.created_at AS created_at_1, groups_1.update_at AS update_at_1, long_live_tokens_1.name AS name_1, long_live_tokens_1.token, long_live_tokens_1.user_id, long_live_tokens_1.id AS id_2, long_live_tokens_1.created_at AS created_at_2, long_live_tokens_1.update_at AS update_at_2 
FROM users LEFT OUTER JOIN groups AS groups_1 ON groups_1.id = users.group_id LEFT OUTER JOIN long_live_tokens AS long_live_tokens_1 ON users.id = long_live_tokens_1.user_id 
WHERE users.email = %(email_1)s]
[parameters: {'email_1': 'jacobmcdonald75023@gmail.com'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

### Mealie Version

1.0.0

### Deployment

TrueNAS

### Additional Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

This fixes more backup-restore-related issues by replacing some of the Alembic ORM stuff with raw SQL.

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes https://github.com/mealie-recipes/mealie/issues/2857

## Special notes for your reviewer:

_(fill-in or delete this section)_

This (and other similar issues) seems to only be an issue when folks are restoring from an older versions (i.e. they make a backup, upgrade, then restore). When restoring on the same version as the backup it seems to be fine.

## Testing

_(fill-in or delete this section)_

Ran my ever-growing test suite of backups and they all restored successfully.

### CODE REVIEW

**Comment by michael-genson** [comment]
Looking into this a little bit more it seems there's still an issue to look into. Will try to find time tomorrow

**Comment by yottabit42** [comment]
> Looking into this a little bit more it seems there's still an issue to look into. Will try to find time tomorrow

How can I pull your branch if I'm deploying using Docker? I guess I just need manually update the files in the image? I was planning to try the restore with your branch, then export again, and attempt to import into RC2.

**Comment by michael-genson** [comment]
That would work, however I'd advise against it since the fix doesn't seem to be fully working. I'd recommend restoring to the version you created your backup from in the meantime.

**Comment by yottabit42** [comment]
I still have that instance. I guess I'll just stay on that old beta until you tell me the fixes are released, and then I'll try to switch to the new instance again.

Thank you!

File: mealie/services/backups_v2/alchemy_exporter.py [hunk 1]
```diff
@@ -39,23 +41,33 @@ def __init__(self, connection_str: str) -> None:
...
             elif isinstance(value, str):
-                if key in AlchemyExporter.look_for_datetime:
-                    data[key] = AlchemyExporter.DateTimeParser(dt=value).dt
-                if key in AlchemyExporter.look_for_date:
-                    data[key] = AlchemyExporter.DateTimeParser(date=value).date
-                if key in AlchemyExporter.look_for_time:
-                    data[key] = AlchemyExporter.DateTimeParser(time=value).time
+                if self.is_uuid(value):
+                    # convert the data to the current database's native GUID type
+                    data[key] = GUID.convert_value_to_guid(value, self.engine.dialect)
```
Comment (michael-genson) [inline mealie/services/backups_v2/alchemy_exporter.py:64]: This is the line that fixes this nightmare

Comment (hay-kot) [reply mealie/services/backups_v2/alchemy_exporter.py:64]: ???? How did you figure this out?

Comment (michael-genson) [reply mealie/services/backups_v2/alchemy_exporter.py:64]: Pain

Comment (michael-genson) [reply mealie/services/backups_v2/alchemy_exporter.py:64]: The actual answer though: I stared at a lot of compiled queries and eventually noticed that all of my GUIDs were lowercase and had no dashes. Then I peeked into my backup data and noticed all of my GUIDs had dashes. Then I threw a hail-mary and bypassed our GUID implementation (see: https://github.com/mealie-recipes/mealie/pull/2859/commits/5cf5abe27966bbc0d2e9e8a13976c2e5b5c234a7) and my restores started working

Looking closer at the [GUID implementation](https://github.com/mealie-recipes/mealie/pull/2859/files#diff-8e4d6d8e87a3a5158219a1321df8a22c77d8ed9538e7bb5a20858fd18ea94885), we keep the dashes in postgres (i.e. we keep the literal GUID representation as a string) and we convert the GUIDs to 32 char hex in sqlite. So sqlite GUIDs, in our implementation, are assumed to have no dashes. But our backup restore doesn't strip them out if they originally came from postgres

**Review by hay-kot — APPROVED** [review]

**Comment by yottabit42** [comment]
I'm very excited for this to be released! How will I know which release version this patch makes it into?

**Comment by boc-the-git** [comment]
It would've already been deployed under the nightly tag.

**Comment by yottabit42** [comment]
Thanks! That's helpful, but since I'm using a packaged version from TrueCharts I won't get the nightly builds. I assume this patch will it make it into RC3 or any later releases? Thanks again!

**Comment by boc-the-git** [comment]
Correct, it'll be in whatever the next formal release is. That should be either rc3 or simply v1.0

**Comment by yottabit42** [comment]
I am now on 1.1.0, but I cannot upload my backup file. When I select the file, nothing ever happens in the UI to show progress of the file uploading, and even after waiting several minutes it does not appear.

**Comment by michael-genson** [comment]
That might be a restriction on your webserver, the default often limits the file size

**Comment by yottabit42** [comment]
I didn't have the problem with the earlier versions of Mealie with the same database backup. It's 110 MB. 

I'm also using the TrueCharts app on TrueNAS so I don't really have much control if what's going on in the container. I might be able to configure it to use an external data store, and then I could try manually copying the database backup into the right location. I'll give that a try tomorrow if I get a chance.

**Comment by michael-genson** [comment]
Yup that was going to be my next suggestion. If you're able to access the mounted filesystem you can add it directly to the backups folder

**Comment by yottabit42** [comment]
Switching to external data storage and moving my backup in myself was successful. The restore worked and I'm back in business!

**Comment by michael-genson** [comment]
Yay! Glad to hear!

**Pull request merged**